dbms/src/Interpreters/ExpressionAnalyzer.h

```diff
@@ -103,7 +103,8 @@ class ExpressionAnalyzer : private boost::noncopyable
     /// Before aggregation:
     bool appendArrayJoin(ExpressionActionsChain & chain, bool only_types);
     bool appendJoin(ExpressionActionsChain & chain, bool only_types);
-    bool appendWhere(ExpressionActionsChain & chain, bool only_types);
+    bool appendPrewhere(ExpressionActionsChain & chain, bool only_types, std::shared_ptr<bool> & remove_filter);
```

Comment:
Cannot understand without comments.
Why we need `std::shared_ptr<bool>`. Looks weird.